frodo.js

```diff
@@ -1,86 +1,83 @@
 /**
```

Comment:
SCRIPT-This function might benefit from using async/await instead of callbacks for better readability.